app/src/main/java/com/mashup/mobalmobal/ui/profile/presenter/ProfileViewModel.kt

```diff
@@ -3,52 +3,143 @@ package com.mashup.mobalmobal.ui.profile.presenter
 import com.funin.base.funinbase.base.BaseViewModel
 import com.funin.base.funinbase.extension.rx.subscribeWithErrorLogger
 import com.funin.base.funinbase.rx.schedulers.BaseSchedulerProvider
-import com.mashup.mobalmobal.constant.Constants
-import com.mashup.mobalmobal.data.sharedpreferences.MobalSharedPreferencesImpl
-import com.mashup.mobalmobal.ui.profile.data.dto.toProfileItems
+import com.mashup.base.extensions.combineLatest
+import com.mashup.mobalmobal.R
+import com.mashup.mobalmobal.data.dto.PostDto
+import com.mashup.mobalmobal.data.dto.UserDto
+import com.mashup.mobalmobal.data.repository.UserRepository
+import com.mashup.mobalmobal.ui.profile.data.dto.MyDonateDto
+import com.mashup.mobalmobal.ui.profile.data.dto.MyPostDto
 import com.mashup.mobalmobal.ui.profile.data.repository.ProfileRepository
 import com.mashup.mobalmobal.ui.profile.domain.model.ProfileItem
+import com.mashup.mobalmobal.ui.profile.domain.model.toProfileItem
+import com.mashup.mobalmobal.ui.profile.domain.model.toProfileItems
+import com.mashup.mobalmobal.ui.profile.domain.model.toSummaryProfileItem
 import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.zipWith
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.PublishSubject
-import java.io.IOException
 import javax.inject.Inject
 
 @HiltViewModel
 class ProfileViewModel @Inject constructor(
     schedulerProvider: BaseSchedulerProvider,
+    userRepository: UserRepository,
     private val profileRepository: ProfileRepository
 ) : BaseViewModel(schedulerProvider) {
 
+    companion object {
+        const val STATUS_DONATION_BEFORE = "BEFORE"
+        const val STATUS_DONATION_INPROGRESS = "IN_PROGRESS"
+        const val STATUS_DONATION_EXPIRED = "EXPIRED"
+    }
+
     private val _toastSubject: PublishSubject<String> = PublishSubject.create()
     val toastSubject get() = _toastSubject
 
-    private val _loadingSubject: PublishSubject<Boolean> = PublishSubject.create()
-    val loadingSubject get() = _loadingSubject
+    val userNickSubject: Observable<String> get() = _userDtoSubject.map { it.nickname }
```

Comment:
```suggestion
    val userNickSubject: Observable<String> = _userDtoSubject.map { it.nickname }
```
이렇게 안하시고 get() = 붙이신 이유가 있으신가요?

Reply:
요건 get을 붙이지 않으면 type 매치가 안되서 에러가 되는데 외부에서 옵저빙하는 userNickSubject는 원시타입으로 명시해주고 싶어서 Observable로 선언했습니다.